pkg/policy/api/rule_validation.go

```diff
@@ -381,6 +381,41 @@ func (pr *PortRule) sanitize(ingress bool) error {
 		}
 	}
 
+	listener := pr.Listener
+	if listener != nil {
+		if listener.Name == "" {
+			return fmt.Errorf("Listener must have name")
+		}
+		if listener.EnvoyConfig == nil {
+			return fmt.Errorf("Listener must have envoyConfig (%s)",
+				listener.Name)
+		}
+		if listener.EnvoyConfig.Name == "" {
+			return fmt.Errorf("Listener envoyConfig must have name (%s)", listener.Name)
+		}
+
+		switch listener.EnvoyConfig.Kind {
+		case "CiliumEnvoyConfig", "CiliumClusterwideEnvoyConfig", "":
+		default:
+			return fmt.Errorf("Listener envoyConfig.kind must be CiliumEnvoyConfig, CiliumClusterwideEnvoyConfig, or left empty (%s)", listener.Name)
+		}
+
+		// For now we have only tested custom listener support on the egress path.
+		// TODO (jrajahalme): Lift this limitation once proper testing has been done on the
+		// ingress path.
+		if ingress {
+			return fmt.Errorf("Listener is not allowed on ingress (%s)", listener.Name)
+		}
+		// There is no quarantee that Listener will support Cilium policy enforcement.
+		// Even now proxylib-based enforcement (e.g, Kafka) may work, but has not been
+		// tested.
+		// TODO (jrajahalme): Lift this limitation for proxylib based parsers if needed and
+		// when tested.
+		if !pr.Rules.IsEmpty() {
```

Comment:
Couple of TODOs here. Do you intend to implement in this PR? Apply follow up comment from above if not.